ddapm_test_agent/agent.py

```diff
@@ -652,6 +642,64 @@ async def handle_trace_analyze(self, request: Request) -> web.Response:
         # wait 1s, gather traces and assert tags
         raise NotImplementedError
 
+    @middleware  # type: ignore
+    async def store_request_middleware(self, request: Request, handler: _Handler) -> web.Response:
+        # only store requests for specific endpoints
+        if request.path in self._forward_endpoints:
+            await self._store_request(request)
+
+        headers = CIMultiDict(request.headers)
+
+        if "X-Datadog-Trace-Env-Variables" in headers:
+            var_string = headers.pop("X-Datadog-Trace-Env-Variables")
+            env_vars = {
+                key.strip(): value.strip() for key, value in (pair.split("=") for pair in var_string.split(","))
+            }
+            request["_dd_trace_env_variables"] = env_vars
+
+        if "X-Datadog-Agent-Proxy-Disabled" in headers:
+            request["_proxy_to_agent"] = headers.pop("X-Datadog-Agent-Proxy-Disabled").lower() != "true"
+
+        if "X-Datadog-Proxy-Port" in headers:
+            port = headers.pop("X-Datadog-Proxy-Port")
+            request.app["agent_url"] = update_trace_agent_port(request.app["agent_url"], new_port=port)
+            log.info("Found port in headers, new trace agent URL is: {}".format(request.app["agent_url"]))
+
+        request["_headers"] = headers
+        # Call the original handlerx
+        return await handler(request)
+
+    @middleware  # type: ignore
+    async def request_forwarder_middleware(self, request: Request, handler: _Handler) -> web.Response:
+        if request.path in self._forward_endpoints:
+            # forward the request then call the handler
+            return await self._forward_request_to_agent(request, handler)
+        else:
+            # Call the original handler and do nothing
+            return await handler(request)
+
+    async def _forward_request_to_agent(self, request: Request, handler: _Handler) -> web.Response:
+        """Forward all requests to the agent_url if set."""
+        data = self._request_data(request)
+        headers = request["_headers"]
+        agent_url = request.app["agent_url"]
+        proxy_to_agent = request.get("_proxy_to_agent", True)
```

Comment:
why are we changing the condition for proxying to the agent? Before it was if there was an agent url detected (and then checking the header). 

```
        proxy_to_agent = agent_url != ""

        if "X-Datadog-Agent-Proxy-Disabled" in headers:
            proxy_to_agent = headers.pop("X-Datadog-Agent-Proxy-Disabled").lower() != "true"
```

can we maintain this functionality?

Reply:
This functionality is maintained